src/pages/DashBoard.js

```diff
@@ -1,21 +1,53 @@
 import React from 'react';
 import { auth, db } from "../utils/firebase";
-import { collection, addDoc } from 'firebase/firestore';
+import { collection, addDoc, setDoc, doc } from 'firebase/firestore';
 import { Button, ButtonGroup, Flex, Heading } from '@chakra-ui/react';
 import Auth from '../components/auth';
 import { useNavigate } from 'react-router-dom';
 import {signOut} from 'firebase/auth';
+import { NumberDictionary, adjectives, animals, colors, uniqueNamesGenerator } from 'unique-names-generator';
+import { checkForRoomIDDupes } from '../components/dbCalls';
+import CreateAlert from '../components/CreateAlert';
 
 const DashBoard = () => {
-
   const navigate = useNavigate();
+  const createAlert = CreateAlert();
 
   const handleHostRoom = async () => {
     try {
       const user = auth.currentUser;
 
       if (user) {
-        const roomRef = await addDoc(collection(db, "rooms"), { 
+
+        let randomRoomNumber = Math.floor(Math.random() * 90000) + 10000;
+        let roomID = uniqueNamesGenerator({ 
+          dictionaries: [adjectives, [randomRoomNumber.toString()]],
+          separator: '',
+          style: 'capital'
+        });
+        let check = await checkForRoomIDDupes(roomID);
+        console.log(check);
+        let runningTime = 1;
+
+        while(!check) {
+          runningTime++;
+          if (runningTime > 300) {
+            console.log('timed out');
+            return createAlert('error', 'Timed Out', 'No Available Room Found', 1500);
+          }
+          randomRoomNumber = Math.floor(Math.random() * 90000) + 10000;
```

Comment:
You have the same code in here twice - let's refactor. You can turn it into a function, or probably even just a do while.